src/main/java/in/projecteka/consentmanager/clients/IdentityServiceClient.java

```diff
@@ -55,18 +57,18 @@ public IdentityServiceClient(WebClient.Builder webClientBuilder,
                 .accept(MediaType.APPLICATION_JSON)
                 .body(BodyInserters.fromFormData(formData))
                 .retrieve()
-                .onStatus(httpStatus -> httpStatus.value() == 401 , clientResponse -> clientResponse.bodyToMono(KeyCloakError.class)
+                .onStatus(httpStatus -> httpStatus.value() == 401, clientResponse -> clientResponse.bodyToMono(KeyCloakError.class)
                         .flatMap(keyCloakError -> {
-                            Error.ErrorBuilder errorBuilder = Error.builder().message(keyCloakError.getErrorDescription());
                             String keyCloakErrorValue = keyCloakError.getError();
-                            if (keyCloakErrorValue.equals("1002")) {
-                                errorBuilder.code(ErrorCode.OTP_INVALID);
-                            }else if (keyCloakErrorValue.equals("1003")) {
-                                errorBuilder.code(ErrorCode.OTP_EXPIRED);
-                            }else {
-                                errorBuilder.code(ErrorCode.UNKNOWN_ERROR_OCCURRED);
+                            switch (keyCloakErrorValue) {
+                                case "1002":
+                                    return onInvalidOTP.perform().then(Mono.error(ClientError.invalidOtp()));
```

Comment:
any particular reason we are not throwing a. client error. like Mono.error(ClientError.invalidOtp())

Reply:
It is throwing the Client error only. If you see this `.then(Mono.error(ClientError.invalidOtp()));`
We have `onInvalidOTP.perform()` before throwing the error because we want to insert a new record in db before throwing the error.